File: src/main/java_ex/vladsaif/syncedit/plugin/lang/transcript/psi/TranscriptTokenType.java
package vladsaif.syncedit.plugin.lang.transcript.psi;

import com.intellij.psi.tree.IElementType;
import org.jetbrains.annotations.NotNull;
import vladsaif.syncedit.plugin.lang.transcript.TranscriptLanguage;

public class TranscriptTokenType extends IElementType {
    public static final IElementType TOO_BIG = new TranscriptTokenType("TOO_BIG_LITERAL");

    public TranscriptTokenType(@NotNull String debugName) {
        super(debugName, TranscriptLanguage.getInstance());
    }

    @Override
    public String toString() {
        return "TranscriptTokenType." + super.toString();
    }
}

File: src/main/kotlin/vladsaif/syncedit/plugin/lang/transcript/parser/TranscriptTokenType.kt
package vladsaif.syncedit.plugin.lang.transcript.parser

import com.intellij.psi.tree.IElementType
import vladsaif.syncedit.plugin.lang.transcript.psi.TranscriptLanguage

class TranscriptTokenType(debugName: String) : IElementType(debugName, TranscriptLanguage) {
  override fun toString(): String {
    return "TranscriptTokenType." + super.toString()
  }
}
